### PR TITLE
Clearer indications of when forecast is different from real

### PR DESCRIPTION
**Pull request by Ben-EJ**
Highlight all AQI values
image
Highlight primary AQI values
image
Added +/- and sorted them accordingly
image

### CODE REVIEW

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/GridCss.css
Comment (rstrange-scottlogic) [inline air-quality-ui/src/components/summary-grid/cell/cell-rules/GridCss.css:1]: Minor, but do we need Css in the file name given it's a CSS file?

File: air-quality-ui/src/services/types.ts [hunk 1]
```diff
@@ -9,11 +9,11 @@ export interface PollutantAverageDataDto<T> {
   mean: T
 }
```
Comment (rstrange-scottlogic) [inline air-quality-ui/src/services/types.ts:11]: Maybe I missed it, but why are you exporting these?

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.spec.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
+//import each from 'jest-each'
```
Comment (rstrange-scottlogic) [inline air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.spec.ts:1]: Can this go along with the imported package in package.json?

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export default function colourCell(
+  showAllColoured: boolean,
+  params: Params,
+  lowerLimit: number,
+  upperLimit?: number,
+): boolean {
+  const pollutantType: string = params.column.colId.split('.')[1]
+  if (!params.value) return false
```
Comment (rstrange-scottlogic) [inline air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts:25]: Can we remove the inline returns from here too?

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  } else {
+    isInColourBand = params.value >= lowerLimit && params.value <= upperLimit
+  }
+
+  if (showAllColoured) return isInColourBand
+  if (!params.data.measurements || !params.data.measurements[pollutantType])
+    return false
+
+  const aqiDifference = Math.abs(
+    params.data.forecast[pollutantType].aqiLevel -
```
Comment (rstrange-scottlogic) [inline air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts:43]: Just to check, are these the overall aqiLevels that show in the summary, i.e. the columns on the far left?

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      params.data.measurements[pollutantType].aqiLevel,
+  )
+
+  if (params.data.aqiDifference === '0') return false
+
+  const aqiMatchesDifference =
+    parseInt(params.data.aqiDifference.split('')[1]) === aqiDifference
```
Comment (rstrange-scottlogic) [inline air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts:50]: No need for the string manipulartion, can we use Math.Abs(params.data.aqiDifference) ?

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/CellRules.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
+import colourCell, { Params } from './ColourCell'
+
+export default function cellRules(showAllColoured: boolean) {
+  return {
+    pm2_5: {
+      'cell-very-good': (params: Params) =>
+        colourCell(showAllColoured, params, 0, 10),
+      'cell-good': (params: Params) =>
+        colourCell(showAllColoured, params, 10, 20),
```
Comment (rstrange-scottlogic) [inline air-quality-ui/src/components/summary-grid/cell/cell-rules/CellRules.ts:9]: I have a suspicion we have somewhere in the UI that can calculate the AQI level for a pollutant value somewhere else. If not maybe we should add one as it's a fairly useful thing to have, and then we could rework this to use it?
It feels like we're missing a trick if we're embedding an AQI converter into a specific CSS class selector when we could make a nice reusable function to do it instead.

Happy if you'd rather do that as part of the future refactoring or in a separate PR (this one is pretty big already)

File: air-quality-ui/src/components/summary-grid/table/sorting/SortingAQIComparator.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+export default function sortAQIComparator(
+  valueA: string,
+  valueB: string,
+): number {
+  if (valueA === undefined) {
```
[the remark this answers is not in the document]
Comment (Ben-EJ) [reply air-quality-ui/src/components/summary-grid/table/sorting/SortingAQIComparator.ts:5]: Yeah, we do need string manipulation as we want to sort by symbols as well. Tried it just to make sure, it does not work. : )

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.spec.ts (hunk 2 above)
Comment (Ben-EJ) [reply air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.spec.ts:1]: I am using this module in SortingAQIComparator.spec.ts, so I am unable to remove it from package.json

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts (hunk 4 above)
Comment (Ben-EJ) [reply air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts:43]: No these are the aqi levels for each polutant. : )

File: air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts (hunk 5 above)
Comment (rstrange-scottlogic) [reply air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts:73]: Happy to do, or it didn't work?

**Review by rstrange-scottlogic — APPROVED** [review]

**Pull request merged**